icu4c/source/common/uloc.cpp

```diff
@@ -505,6 +505,7 @@ static const CanonicalizationMap CANONICALIZE_MAP[] = {
     { "uz_UZ_LATN",     "uz_Latn_UZ", NULL, NULL }, /* .NET name */
     { "zh_CHS",         "zh_Hans", NULL, NULL }, /* .NET name */
     { "zh_CHT",         "zh_Hant", NULL, NULL }, /* .NET name */
+    { "zh_CMN",         "cmn", NULL, NULL }, /* registered name */
```

Comment:
CLDR says map cmn to zh. We follow CLDR.
Ideally, we should generate this data, not add one-offs manually.
I commented on the ticket.

Reply:
Well, Ecma 402 wants to follow BCP 47 + IANA language subtag registry.   I added my comments and related ticket references to ICU-7656 . 

@FrankYFTang , you don't need this PR.  My recent PR (merged yesterday) handles this in uloc_tag.cpp.  If you call ```uloc_toLanguageTag(uloc_forLanguageTag ("zh-cmn"))``` (not a real code but conceptually), you'll get back 'cmn'.

Reply:
@jungshik somehow right now uloc_toLanguageTag("zh_CMN") give me "zh" instead "cmn"